pkg/controller/pod/pod_controller.go

```diff
@@ -76,8 +77,23 @@ type ReconcilePolicy struct {
 // Reconcile reads that state of the cluster for a Pod object and makes changes based on the state
 func (r *ReconcilePolicy) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	log.V(1).Info("got a pod event in the controller")
+	instanceOptedIn, err := r.poolManager.IsPodInstanceOptedIn(request.Namespace)
+	if err != nil {
+		// Error reading the object - requeue the request.
+		log.Error(err, "failed to check opt-in selection",
```

Comment:
No need for this, controller-runtime will print the log for us, just do the errors.Wrap thing